napari/_qt/qt_resources/_svg.py

```diff
@@ -0,0 +1,149 @@
+"""
+A Class for generating QIcons from SVGs with arbitrary colors at runtime.
+"""
+from __future__ import annotations
+
+from functools import lru_cache
+
+from qtpy.QtCore import QByteArray, QPoint, QRect, QRectF, Qt
+from qtpy.QtGui import QIcon, QIconEngine, QImage, QPainter, QPixmap
+from qtpy.QtSvg import QSvgRenderer
+
+
+class QColoredSVGIcon(QIcon):
+    """A QIcon class that specializes in colorizing SVG files.
+
+    Parameters
+    ----------
+    path_or_xml : str
+        Raw SVG XML or a path to an existing svg file.  (Will raise error on
+        ``__init__`` if a non-existent file is provided.)
+    color : str, optional
+        A valid CSS color string, used to colorize the SVG. by default None.
+    opacity : float, optional
+        Fill opacity for the icon (0-1).  By default 1 (opaque).
+
+    Examples
+    --------
+    >>> from napari._qt.qt_resources import QColoredSVGIcon
+    >>> from qtpy.QtWidgets import QLabel
+
+    # Create icon with specific color
+    >>> label = QLabel()
+    >>> icon = QColoredSVGIcon.from_resources('new_points')
+    >>> label.setPixmap(icon.colored('#0934e2', opacity=0.7).pixmap(300, 300))
+    >>> label.show()
+
+    # Create colored icon using theme
+    >>> label = QLabel()
+    >>> icon = QColoredSVGIcon.from_resources('new_points')
+    >>> label.setPixmap(icon.colored(theme='light').pixmap(300, 300))
+    >>> label.show()
+    """
+
+    def __init__(
+        self, path_or_xml: str, color: str = None, opacity: float = 1.0
+    ) -> None:
+        from ...resources import get_colorized_svg
+
+        self._svg = path_or_xml
+        colorized = get_colorized_svg(path_or_xml, color, opacity)
+        super().__init__(SVGBufferIconEngine(colorized))
+
+    @lru_cache()
+    def colored(
+        self,
+        color: str = None,
+        opacity: float = 1.0,
+        theme: str = None,
+        theme_key: str = 'icon',
+    ):
+        """Return a new colorized QIcon instance.
+
+        Parameters
+        ----------
+        color : str, optional
+            A valid CSS color string, used to colorize the SVG.  If provided,
+            will take precedence over ``theme``, by default None.
```

Comment:
```suggestion
            will take precedence over `theme`, by default None.
```

Reply:
it was my understanding that when sphinx renders our api docs, it uses ReST (even if we're using jupyter-book)... so all of these single backticks will render as _italicized_, rather than `code`.  See, for example, all of the italicized "_True_" in [this docstring](https://napari.org/docs/dev/api/napari.layers.Image.html) ... whereas [this function](https://napari.org/docs/dev/api/napari.utils.html#napari.utils.resize_dask_cache) uses double backticks and renders correctly.  Thoughts?

Reply:
italicised vs verbatim is a style discussion. Double backticks vs single backticks is semantically distinctions.
If you use single backticks sphinx should be able to infer that `theme` is a parameter (potentially adding scrolling anchors to hat parameter); not if you use double. 

If italic is wrong, then IMHO the CSS/Rendering/Sphinx should be changed, not the docstrings.

I personally care as the terminal renderer/docstring fixer I write need the correct semantic to find typos. It will ignore double-backticks as it's supposed to be verbatim. (well i could /should inspect verbatim and tell user when they might be wrong... but that's another discussion).

If you still like double backticks I'm fine but you should likely change `` `theme_key` `` to double backticks line 71.

Reply:
(note, I personally also hate all those things in italic in this theme and in numpy documentation in general, and agree with you that monospace font is better suited, but I believe this to not be relevant for the argument)

Reply:
I 100% agree with your arguments here.  Semantics and syntax are definitely being muddied here, and that poses a problem for the larger goal of having truly parseable docstrings.  I looked into it a bit and it looks to be an issue that comes up in a lot of repos.  Looks like our current sphinx docs will render single backticks as `<cite>content</cite>` (whereas true italics is `<em>` as expected) and double backticks with the `<span class="pre">content</span>`, so we can definitely attack this on the CSS side.  I'll make a new issue for this, since this is all over the repo and much larger than this PR

Reply:
created #2286